### ISSUE TITLE
unchecked_time_subtraction triggers when underflow is impossible

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16499]
### Summary

`unchecked_time_subtraction` triggers in cases where underflow is statically impossible.

### Lint Name

`unchecked_time_subtraction`

### Reproducer

```rust
pub fn m1() -> Duration {
    Duration::from_millis(2) - Duration::from_millis(1)
}
```

https://godbolt.org/z/ze67rjKEv

#### Expected behavior

The lint should not trigger in this case.

Both operands are compile-time constants, and it is provable that the left-hand Duration is greater than the right-hand one. Underflow cannot occur.

#### Actual behaviour

Clippy warns and suggests replacing the subtraction with `checked_sub(...).unwrap()`.

```
warning: unchecked subtraction of a `Duration`
 --> <source>:5:5
  |
5 |     Duration::from_millis(2) - Duration::from_millis(1)
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `Duration::from_millis(2).checked_sub(Duration::from_millis(1)).unwrap()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unchecked_time_subtraction
  = note: requested on the command line with `-W clippy::unchecked-time-subtraction`

warning: 1 warning emitted
```
